MV3 host permission install prompt (#34094)
mdn/content: files/en-us/mozilla/add-ons/webextensions/manifest.json/host_permissions/index.md

### Requested permissions and user prompts

> Users can [grant or revoke host permissions](https://support.mozilla.org/en-US/kb/extensions-button?_gl=1*bp9tzb*_ga*OTg4Nzg5MzUzLjE1NTg3MzE4Mzg.*_ga_X4N05QV93S*MTcxODg3NTc5NC40Ni4wLjE3MTg4NzU3OTQuMC4wLjA.#w_website-permissions) on an ad hoc basis. Therefore., most browsers treat `host_permissions` as optional. 

Reviewer: ```suggestion
Users can [grant or revoke host permissions](https://support.mozilla.org/en-US/kb/extensions-button#w_website-permissions) on an ad hoc basis. Therefore., most browsers treat `host_permissions` as optional.
```

On installation, when you request permissions using this key:

> - Until Firefox 126, requested host permissions weren't displayed in the install prompt. From Firefox 127, host permissions listed in `host_permissions` and `content_scripts` are displayed in the install prompt. However, if an extension update requests new host permissions, these are not shown to the user. See ([Firefox bug 1893232](https://bugzil.la/1893232)).

Reviewer: ```suggestion
- Until Firefox 126, requested host permissions of Manifest Version 3 extensions weren't displayed in the install prompt. From Firefox 127, host permissions listed in `host_permissions` and `content_scripts` are displayed in the install prompt. However, if an extension update requests new host permissions, these are not shown to the user. See ([Firefox bug 1893232](https://bugzil.la/1893232)).
```

Small edit to emphasize MV3. `host_permissions` is MV3-only, but the general advice here applies to any host permission, and this is the only documentation that we have on this.

Author: ```suggestion
- Until Firefox 126, a Manifest V3 extension's requested host permissions weren't displayed in the install prompt. From Firefox 127, host permissions listed in `host_permissions` and `content_scripts` are displayed in the install prompt. However, if an extension update requests new host permissions, these are not shown to the user. See ([Firefox bug 1893232](https://bugzil.la/1893232)).
```

Most browsers treat `host_permissions` as optional. If you request permissions using this key, users _may_ get prompted to grant those permissions during installation. As of June 2023, Safari, Firefox, and some Chromium-based browsers don't prompt the user during installation.

> > **Note:** From Firefox 127. `host_permissions` for Manifest V3 extensions is considered a required permission and displayed to users during installation. However, if an extension update grants new host permissions, these are not shown to the user. See ([Firefox bug 1893232](https://bugzil.la/1893232))

Reviewer: ```suggestion
> **Note:** Until Firefox 126, Firefox did not include requested host permissions of Manifest Version 3 extensions in the install prompt. Starting from Firefox 127, host permissions listed in `host_permissions` and `content_scripts` do appear in the install prompt. However, if an extension update requests new host permissions, these are not shown to the user. See ([Firefox bug 1893232](https://bugzil.la/1893232)).
```

On wording: "required" is not accurate.

The previous paragraph tries to emphasize that host_permissions are optional, and that is still the case: users can revoke them at any time.

The wording as written is technically true but potentially overstating the unreliability of them.

In practice:
- Chrome shows the permissions in the install prompt.
- Firefox 127 shows the permissions in the install prompt. Up until Firefox 126, host permissions were not shown in the install prompt. Starting from 127, they are. However, when a new permission is added in an update, Firefox will not show any prompt for it. Extensions who rely on the new host permissions are encouraged to use the permissions.contains API after install (runtime.onInstalled) to check whether the permission has been granted and show an onboarding page if needed.

> Most browsers treat `host_permissions` as optional. If you request permissions using this key, users _may_ get prompted to grant those permissions during installation. As of June 2023, Safari, Firefox, and some Chromium-based browsers don't prompt the user during installation.
> 
> > **Note:** From Firefox 127. `host_permissions` for Manifest V3 extensions is considered a required permission and displayed to users during installation. However, if an extension update grants new host permissions, these are not shown to the user. See ([Firefox bug 1893232](https://bugzil.la/1893232))
> 
> Users can also grant or revoke host permissions on an ad hoc basis. For example, in Firefox, users can do this using the [extensions panel](https://blog.mozilla.org/addons/2022/11/17/unified-extensions-button-and-how-to-handle-permissions-in-manifest-v3/).

Author: @Rob--W is this better?
```suggestion
Users can [grant or revoke host permissions](https://support.mozilla.org/en-US/kb/extensions-button?_gl=1*bp9tzb*_ga*OTg4Nzg5MzUzLjE1NTg3MzE4Mzg.*_ga_X4N05QV93S*MTcxODg3NTc5NC40Ni4wLjE3MTg4NzU3OTQuMC4wLjA.#w_website-permissions) on an ad hoc basis. Therefore., most browsers treat `host_permissions` as optional. 

On installation, when you request permissions using this key:

- Until Firefox 126, requested host permissions weren't displayed in the install prompt. From Firefox 127, host permissions listed in `host_permissions` and `content_scripts` are displayed in the install prompt. However, if an extension update requests new host permissions, these are not shown to the user. See ([Firefox bug 1893232](https://bugzil.la/1893232)).
- Chrome displays the permissions in the install prompt.
- Safari doesn't display requested host permissions in the install prompt.
```
